packages/ripple/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { getRippleGlobalNamespace } from './getRippleGlobalNamespace';
+import { Variant, colorMap, Mode } from './utils';
+
+const TRANSITION_TIME = 300;
+const RIPPLE_NAMESPACE = getRippleGlobalNamespace();
+
+interface Options {
+  variant: Variant;
+  darkMode: boolean;
+}
+
+let globalOptions: Options = { darkMode: false, variant: Variant.Default };
+
+function rippleEvent(event: MouseEvent) {
+  if (
+    // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+    RIPPLE_NAMESPACE.registeredRippleElements[event.target]
+  ) {
+    createRippleEffect(event);
+  }
+}
+
+export function registerRipple(node: HTMLElement, options: Options) {
+  if (!RIPPLE_NAMESPACE) {
+    return;
+  }
+
+  globalOptions = options;
```

Comment:
Seems less then great that you can only change global options when registering an element.

Reply:
You can update the global options by calling register element inside of a hook that has variant, etc. as dependents

Reply:
Right, but you need to include a `node` to modify options.

Reply:
Updated the entire implementation around this, but we would need a node to modify the options otherwise we won't know whose options to modify. The logic around this is a lot cleaner now